client/ayon_silhouette/api/plugin.py

```diff
@@ -31,23 +33,37 @@ def cache_instance_data(shared_data):
 
         instance_ids = {AYON_INSTANCE_ID}
         for node in session.nodes:
-            data = lib.read(node)
-            if not data:
+            instances_data_by_uuid = lib.read(node, key=INSTANCES_DATA_KEY)
+            if not instances_data_by_uuid:
                 continue
 
-            if data.get("id") not in instance_ids:
-                continue
+            for uuid, data in instances_data_by_uuid.items():
```

Comment:
I would not be specific about `uuid`.
```suggestion
            for instance_id, data in instances_data_by_id.items():
```

Reply:
I've thought about this - but technically, they are not the instance ids ;)

The publish instance id needs to be completely  unique across the scene.
The ids stored in the 'dict' on the node is the uuid unique within that node's data - it's not a scene wide instance id.

The publish instance id is `node.id` + this `uuid` in the dict so that if you were to duplicate the node itself, the internal data uuid would still be overlapping but the prepended `node.id` would still make it unique to the publisher system otherwise, the duplicated node entries would overlap in `instance.id`.

Thoughts?